legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/nodes/ExecutionNodeExecutor.java

```diff
@@ -230,6 +236,26 @@ public Result visit(AllocationExecutionNode allocationExecutionNode)
         {
             result = new ConstantResult(((Map<?, ?>) ((ConstantResult) result).getValue()).get("value"));
         }
+        // TODO HSO: Right place for this? Other streaming result types?
```

Comment:
I think each result set should implement (or throw if not supported) the `realizeInMemory` method.  Then, this code will be encapsulated on the right place.

One consideration tho is the limit we might want to set when realizing in memory like we do for relational results.